src/v/cluster/tx_gateway_frontend.cc

```diff
@@ -258,9 +258,15 @@ ss::future<try_abort_reply> tx_gateway_frontend::do_try_abort(
                     try_abort_reply{.aborted = true, .ec = tx_errc::none});
               }
               auto tx_id = tx_id_opt.value();
-              return stm->get_tx_lock(tx_id)->with(
-                [&self, stm, tx_id, pid, tx_seq, timeout]() {
-                    return self.do_try_abort(stm, tx_id, pid, tx_seq, timeout);
+              return stm->get_tx_lock(tx_id)
+                ->with(
+                  timeout,
```

Comment:
* is this a long term solution?
* if we know what the deadlock is, can we avoid it?
* if the deadlock is common, this delay will be incurred. who eats this timeout cost on deadlock?
    * how long is the timeout
    * how common is the deadlock?

Reply:
thanks for the heads up, I avoided grabbing lock during try_abort so now this deadlock is impossible